src/buffer/out/textBuffer.hpp

```diff
@@ -161,10 +161,16 @@ class TextBuffer final
                               const std::wstring_view fontFaceName,
                               const COLORREF backgroundColor);
 
+    struct PositionInformation
+    {
+        short mutableViewportTop{ 0 };
+        short visibleViewportTop{ 0 };
+    };
+
     static HRESULT Reflow(TextBuffer& oldBuffer,
                           TextBuffer& newBuffer,
                           const std::optional<Microsoft::Console::Types::Viewport> lastCharacterViewport,
-                          std::optional<short>& oldViewportTop);
+                          std::optional<std::reference_wrapper<PositionInformation>> positionInfo);

```

Comment:
```suggestion
                          std::optional<std::reference_wrapper<PositionInformation>> positionInfo = std::nullopt);
```
I think you can do it for the `lastCharacterViewport` parameter too.

Then the call you have in screenInfo doesn't have two `std::nullopt`s passed through.

Up to you tho